src/terminal/parser/stateMachine.cpp

```diff
@@ -728,14 +726,14 @@ void StateMachine::_ActionDcsDispatch(const wchar_t wch)
 
     const auto success = _SafeExecute([=]() {

```

Comment:
So this lambda, combined with "the important bit" below, is like, the actual PR here?

Reply:
Yeah. I apologize for the complete mess of this PR. ☹️
Initially I thought this would be easy, but then the sunken cost fallacy hit me. ☹️☹️